src/components/Schemas/Schemas.jsx

```diff
@@ -0,0 +1,181 @@
+  const [containerHeight, setContainerHeight] = useState('auto');
+
+  const inView = useInView(containerRef, { amount: 'some' });
+  const [viewed, setViewed] = useState(inView);
+  const data = Object.entries(schemas).map(mapSchemaDto);
+  const containerVariants = {
+    hidden: {},
+    show: {
+      transition: {
+        staggerChildren: 0.025,
+      },
+    },
+    selected: {},
+    idle: {},
+  };
+
+  // This controls the steps of animation for the cards.
+  // It goes from: undefined -> show -> idle | selected-id
+  const animate = useMemo(() => {
+    if (selectedSchema) return `selected-${selectedSchema.id}`;
+    if (viewed) return 'idle';
+    return inView ? 'show' : undefined;
+  }, [selectedSchema, inView, viewed]);
+  console.log(animate);
+
+  useEffect(() => {
+    if (viewed) return;
+    // The first time user sees the content we show the animation and set as viewed.
+    inView && setViewed(true);
+  }, [inView, viewed]);
+
+  useEffect(() => {
+    const observer = new ResizeObserver((entries) => {
+      setContainerHeight(entries[0].contentRect.height);
+    });
+
+    observer.observe(containerRef.current);
+
+    return () => {
+      observer.disconnect();
+    };
+  }, []);
+
+  return (
+    <motion.div
+      className={styles.container}
+      animate={{ height: containerHeight, transition: { duration: 0.2 } }}
+      layout
+    >
+      <motion.div ref={containerRef}>
+        <motion.div
+          variants={containerVariants}
+          initial='hidden'
+          animate={animate}
+          whileInView='show'
+          viewport={{ once: true, margin: '-50%' }}
+        >
+          <input
+            className={styles.search}
+            placeholder='Search for a schema e.g Email Credential'
+            onChange={(e) => setFilter(e.target.value)}
+          />
+          <motion.div className={styles['container-grid']}>
+            <AnimatePresence>
+              {data
+                .filter(filterBySchemaName(filter))
+                .sort(sortBySchemaName)
+                .map(([key, schema]) => (
+                  <SchemaCard
+                    key={key}
+                    schema={schema}
+                    onSchemaSelect={setSelectedSchema}
+                  />
+                ))}
+            </AnimatePresence>
+          </motion.div>
+        </motion.div>
+        <AnimatePresence initial={false}>
+          {selectedSchema && (
+            <SchemaDetails
+              schema={selectedSchema}
+              onRequestClose={() => setSelectedSchema(null)}
+            />
+          )}
+        </AnimatePresence>
+      </motion.div>
+    </motion.div>
+  );
+}
+
+export default function Schemas(props) {
+  const [isLoading, setIsloading] = useState(true);
+  const [schemas, setSchemas] = useState(null);
+
+  // Numbers of schema is for adding the number of renders for SchemaCardSkeleton to have a good feedback for user.
+  const NUMBER_OF_SCHEMAS = 28;
```

Comment:
Why are setting this as a constant? Is it a UI decision so only 28 tiles show up in the component? I feel like probably want to show all of them and allow for scrolling? No idea how difficult that would be though.

Reply:
> Why are setting this as a constant? Is it a UI decision so only 28 tiles show up in the component? I feel like probably want to show all of them and allow for scrolling? No idea how difficult that would be though.

This is just for loading the skeletons, we use what is on `schemas` state.